components/ws-manager-mk2/controllers/timeout_controller.go

```diff
@@ -202,6 +202,19 @@ func (r *TimeoutReconciler) isWorkspaceTimedOut(ws *workspacev1.Workspace) (reas
 			}
 			return decide(start, timeout, activityNone)
 		} else if isClosed {
+			reason := func() string {
+				afterClosed := timeouts.AfterClose
+				if customClosedTimeout := ws.Spec.Timeout.ClosedTimeout; customClosedTimeout != nil {
```

Comment:
the custom closed timeout wasn't working for me, it was `0s` in the workspace CRD, but the workspace still timed out 2 minutes after closing it with message `workspace timed out after after being closed (00h02m) took longer than 00h02m`.

I'm thinking it might be because the `ClosedTimeout` field has the `omitempty` tag, could it be that a 0s duration is interpreted as empty and therefore is `nil` here?